src/chess/bitboard.h

```diff
@@ -190,40 +190,46 @@ class Move {
  public:
   enum class Promotion : std::uint8_t { None, Queen, Rook, Bishop, Knight };
   Move() = default;
-  Move(BoardSquare from, BoardSquare to) : from_(from), to_(to) {}
+  Move(BoardSquare from, BoardSquare to)
+      : data_(to.as_int() + (from.as_int() << 6)) {}
   Move(BoardSquare from, BoardSquare to, Promotion promotion)
-      : from_(from), to_(to), promotion_(promotion) {}
-  Move(const std::string& str, bool black = false);
-  Move(const char* str, bool black = false) : Move(std::string(str), black) {}
+      : data_(to.as_int() + (from.as_int() << 6) +
+              (static_cast<uint8_t>(promotion) << 12)) {}
+  Move(const std::string &str, bool black = false);
+  Move(const char *str, bool black = false) : Move(std::string(str), black) {}
+
+  BoardSquare GetTo() const { return {data_ & kToMask}; }
+  BoardSquare GetFrom() const { return {(data_ & kFromMask) >> 6}; }
+  Promotion GetPromotion() const {
+    return (Promotion)((data_ & kPromoMask) >> 12);
```

Comment:
`{(data_ & kPromoMask) >> 12}`

Reply:
doesn't work, as I commented on the earlier one
../../src/chess/bitboard.h:204:39: error: cannot convert ‘<brace-enclosed initializer list>’ to ‘lczero::Move::Promotion’ in return
     return {(data_ & kPromoMask) >> 12};